docs/internals/contributing/writing-code/coding-style.txt

```diff
@@ -214,20 +214,150 @@ Imports
 Template style
 ==============
 
-* In Django template code, put one (and only one) space between the curly
-  brackets and the tag contents.
+Follow the below rules in Django template code.
+
+* The ``{% extends %}`` should be the first non-comment line.
+
+  Do this:
+
+  .. code-block:: html+django
+
+      {% extends "base.html" %}
+
+      {% block content %}
+        <h1 class="font-semibold text-xl">
+          {{ pages.title }}
+        </h1>
+      {% endblock content %}
+
+  Or this:
+
+  .. code-block:: html+django
+
+      {# This is a comment #}
+      {% extends "base.html" %}
+
+      {% block content %}
+        <h1 class="font-semibold text-xl">
+          {{ pages.title }}
+        </h1>
+      {% endblock content %}
+
+  Don't do this:
+
+  .. code-block:: html+django
+
+      {% load i18n %}
+      {% extends "base.html" %}
+
+      {% block content %}
+        <h1 class="font-semibold text-xl">
+          {{ pages.title }}
+        </h1>
+      {% endblock content %}
+
+* Put one (and only one) space between the curly brackets and the tag contents.
+
+  Do this:
+
+  .. code-block:: html+django
+
+      {{ user }}
+
+  Don't do this:
+
+  .. code-block:: html+django
+
+      {{user}}
+
+* The ``{% load ... %}`` should list items in alphabetical order.
+
+  Do this:
+
+  .. code-block:: html+django
+
+      {% load i18n l10 tz %}
+
+  Don't do this:
+
+  .. code-block:: html+django
+
+      {% load l10 i18n tz %}
+
+* Put one (and only one) space between the curly brackets, the ``%``, and the
+  tag contents.
 
   Do this:
 
   .. code-block:: html+django
 
-      {{ foo }}
+      {% load humanize %}
 
   Don't do this:
 
   .. code-block:: html+django
 
-      {{foo}}
+      {%load humanize%}
+
+* Each ``{% block %}`` should be named with the  ``{% endblock %}`` having the
+  matching name.
```

Comment:
Using names on the same line is a bit verbose?

```suggestion
* Put the ``{% block %}`` tag name in the ``{% endblock %}`` tag if it is not on the same line.
```

Reply:
I think it's still worth doing, as `block` are intended to be extended in templates.

Reply:
But the same syntax is used for filling in. `{% block title %}Something{% endblock %}` is clear enough to me.

Reply:
Agreed, updated.